tensorflow/python/eager/def_function.cc

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <Python.h>
+
+#include <deque>
+
+#include "pybind11/pybind11.h"
+#include "pybind11/stl_bind.h"
+
+struct CallCounter;  // Forward declaration.
+
+namespace tensorflow {
+
+namespace py = pybind11;
+
+// Class keeping track of how many recent calls triggered tracing.
+struct CallCounter {
```

Comment:
nit: This is a bit more complex than POD, let's use class instead of struct. https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes